### ISSUE TITLE
[BUG] Creation of RecurringJob in Quarkus native image

### ISSUE THREAD

**Issue by terziisk** [jobrunr/jobrunr#879]
### JobRunr Version

6.3.2

### JDK Version

quarkus-micro-image:1.0, Java 17

### Your SQL / NoSQL database

PostgreSQL

### What happened?

Locally `scheduleRecurrently` works fine. But deployed on the environment native build of Quarkus application we faced the issue, that `SerializedLambdaConverter` cannot find method `writeReplace` in the passed Labmda function. 

The problem likely stems from the reflection-related limitations of native images in GraalVM. When compiling to a native image, not all reflection features are enabled by default, which may lead to runtime errors. So we tried to registrer all of the engaged classes for reflection,  but it didn't help.

### How to reproduce?

```java
@ApplicationScoped
@RequiredArgsConstructor
@RegisterForReflection
public class JobCreatorService {

  private final AppProperties appProperties;
  private final JobScheduler jobScheduler;

  public String createRecurringJob(final CreateRecurringJobRequest request) {
    return BackgroundJob.<RestExecutor>scheduleRecurrently(
      request.getUniqueName(),
      request.getCron(),
      request.getTimeZone().orElse(appProperties.defaultTimeZone()),
      myService -> myService.callExternalRestApi(request.getRequestDetails(), request.getUniqueName())
    );
  }
}

import org.jobrunr.jobs.details.CachingJobDetailsGenerator;
import org.jobrunr.jobs.details.JobDetailsAsmGenerator;
import org.jobrunr.jobs.details.SerializedLambdaConverter;
import org.jobrunr.scheduling.JobScheduler;
import org.jobrunr.utils.StringUtils;
import org.jobrunr.utils.reflection.ReflectionUtils;

import io.quarkus.runtime.annotations.RegisterForReflection;

@RegisterForReflection(
  ignoreNested = false,
  serialization = true,
  targets = {
    CachingJobDetailsGenerator.class,
    JobDetailsAsmGenerator.class,
    SerializedLambdaConverter.class,
    ReflectionUtils.class,
    StringUtils.class,
    JobScheduler.class
  },
  lambdaCapturingTypes = {
    "org.jobrunr.scheduling.JobLambda"
  }
)
public class ReflectionConfiguration {

}

Dockerfile:
FROM quay.io/quarkus/quarkus-micro-image:1.0
WORKDIR /work/
RUN chown 1001 /work \
    && chmod "g+rwX" /work \
    && chown 1001:root /work
COPY --chown=1001:root build/*-runner /work/application

EXPOSE 8080 8000
USER 1001

CMD ["./application", "-Dquarkus.http.host=0.0.0.0"]
```

### Relevant log output

```shell
Cause:
org.jobrunr.JobRunrException : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace)
{
  "class": "org.jobrunr.jobs.details.SerializedLambdaConverter",
  "method": "toSerializedLambda",
  "line": 32
},
{
  "class": "org.jobrunr.jobs.details.JobDetailsAsmGenerator",
  "method": "toJobDetails",
  "line": 27
},
{
  "class": "org.jobrunr.jobs.details.CachingJobDetailsGenerator$CacheableJobDetails",
  "method": "getJobDetails",
  "line": 94
},
{
  "class": "org.jobrunr.jobs.details.CachingJobDetailsGenerator",
  "method": "toJobDetails",
  "line": 45
},
{
  "class": "org.jobrunr.scheduling.JobScheduler",
  "method": "scheduleRecurrently",
  "line": 581
},
{
  "class": "org.jobrunr.scheduling.BackgroundJob",
  "method": "scheduleRecurrently",
  "line": 550
},
{
  "class": "com.domain.service.JobCreatorService",
  "method": "createRecurringJob",
  "line": 26
},
{
  "class": "com.domain.service.JobCreatorService_ClientProxy",
  "method": "createRecurringJob"
}
....

Root cause:
java.lang.NoSuchMethodException
{
  "class": "java.lang.Class",
  "method": "checkMethod",
  "line": 1038
},
{
  "class": "java.lang.Class",
  "method": "getDeclaredMethod",
  "line": 1128
},
{
  "class": "org.jobrunr.jobs.details.SerializedLambdaConverter",
  "method": "toSerializedLambda",
  "line": 28
},
{
  "class": "org.jobrunr.jobs.details.JobDetailsAsmGenerator",
  "method": "toJobDetails",
  "line": 27
},
{
  "class": "org.jobrunr.jobs.details.CachingJobDetailsGenerator$CacheableJobDetails",
  "method": "getJobDetails",
  "line": 94
},
{
  "class": "org.jobrunr.jobs.details.CachingJobDetailsGenerator",
  "method": "toJobDetails",
  "line": 45
},
{
  "class": "org.jobrunr.scheduling.JobScheduler",
  "method": "scheduleRecurrently",
  "line": 581
},
{
  "class": "org.jobrunr.scheduling.BackgroundJob",
  "method": "scheduleRecurrently",
  "line": 550
},
{
  "class": "com.domain.service.JobCreatorService",
  "method": "createRecurringJob",
  "line": 26
},
{
  "class": "com.domain.service.JobCreatorService_ClientProxy",
  "method": "createRecurringJob"
}
....
```

**Comment by rdehuyss** [comment]
You're on the bleeding edge of JobRunr/GraalVM/Quarkus here.

For native image to work, you need to create your jobs using `JobRequest` / `JobRequestHandler`. At this time, I did not yet look into the newer Serialization Features of GraalVM.

**Comment by rdehuyss** [comment]
P.s.: I think you're missing something on your 

```
@ApplicationScoped
@RequiredArgsConstructor
@RegisterForReflection(serialization = true) // this is the class where you need to enable serialisation as the lambda is inside this class (perhaps also on RestExecutor)
public class JobCreatorService {

}
```

**Comment by terziisk** [comment]
@rdehuyss The approach using `JobRequest / JobRequestHandler` - works fine. Thanks! 

With this approach we faced another issue - if a job failed, then on retry we were receiving serialization exception on `selectJobsScheduledBefore`.
We had such dependencies:
`implementation 'io.quarkus:quarkus-jackson'
implementation 'com.fasterxml.jackson.core:jackson-databind:2.16.0'`
Then we changed it to 
`implementation 'io.quarkus:quarkus-jackson'
implementation 'io.quarkus:quarkus-resteasy-jackson'`
and this problem was solved. This information may help somebody because the Installation guide recommended some other libraries.
